src/server/controllers/UserTransaction.controller.ts

```diff
@@ -0,0 +1,175 @@
+import { GetManyUserTransactionDTO } from '@/lib/models/dto/GetManyUserTransaction.dto';
+
+class UserTransactionController {
+  async create(req: NextApiRequest, res: NextApiResponse): Promise<void> {
+    const body: CreateUserTransactionDTO = req.body;
+    validateSchema<CreateUserTransactionDTO>(createUserTransactionDtoSchema, body);
+
+    const response = await prisma.userTransaction.create({
+      data: {
+        contractHash: body.contractHash,
+        amount: body.amount,
+        userId: body.userId,
+      },
+    });
+
+    if (!response) {
+      throw new InternalServerErrorException('UserTransaction could not be created');
+    }
+
+    res.status(HttpResponseCodesEnum.CREATED).json(resolveInfinitePaginationResponse(response));
+  }
+
+  async deleteById(req: NextApiRequest, res: NextApiResponse): Promise<void> {
+    const query: IdDTO = req.query;
+    validateSchema<IdDTO>(idDtoSchema, query);
+
+    const response: null | UserTransaction = await prisma.userTransaction.delete({
+      where: {
+        id: Number(query.id),
+      },
+    });
+
+    if (!response) {
+      throw new NotFoundException(`User Transaction, with id ${query.id}, not found`);
+    }
+
+    res.status(HttpResponseCodesEnum.OK).json(resolveInfinitePaginationResponse(response));
+  }
+
+  async getByHash(req: NextApiRequest, res: NextApiResponse): Promise<void> {
+    const query: HashDTO = req.query;
+    validateSchema<HashDTO>(hashDtoSchema, query);
+
+    const response: null | UserTransaction = await prisma.userTransaction.findUnique({
+      where: {
+        contractHash: query.hash,
+      },
+    });
+
+    if (!response) {
+      throw new NotFoundException(`User, with hash ${query.hash}, not found`);
+    }
+
+    res.status(HttpResponseCodesEnum.OK).json(resolveInfinitePaginationResponse(response));
+  }
+
+  async getById(req: NextApiRequest, res: NextApiResponse): Promise<void> {
+    const query: IdDTO = req.query;
+    validateSchema<IdDTO>(idDtoSchema, query);
+
+    const response: null | UserTransaction = await prisma.userTransaction.findUnique({
+      where: {
+        id: Number(query.id),
+      },
+    });
+
+    if (!response) {
+      throw new NotFoundException(`User Transaction, with id ${query.id}, not found`);
+    }
+
+    res.status(HttpResponseCodesEnum.OK).json(resolveInfinitePaginationResponse(response));
+  }
+
+  async getMany(req: NextApiRequest, res: NextApiResponse): Promise<void> {
+    const query: GetManyUserTransactionDTO = req.query as unknown as GetManyUserTransactionDTO;
+    const pagination: InfinitePaginationType = resolveInfinitePagination(req.query);
+    validateSchema<GetManyUserTransactionDTO>(getManyUserTransactionDtoSchema, query);
+
+    const response = await prisma.userTransaction.findMany({
+      where: resolveBulkArgs<Prisma.UserTransactionWhereInput>([
+        {
+          key: 'id',
+          value: query?.ids,
+        },
+        {
+          key: 'amount',
```

Comment:
Since its a bulk search arg, we can write the arg in plural, ex: for amount, write amounts